src/app/src/mainwindow.cpp

```diff
@@ -1032,13 +1004,28 @@ void MainWindow::on_disableBackgroundRecording_stateChanged(int state) {
     } else {
         spdlog::info("Background subtraction disabled");
     }
+    checkStartAcqRequirements(false, false);
+}
+
+void MainWindow::checkStartAcqRequirements(bool logDriveCheck, bool logFrameRateAndDurCheck) {
```

Comment:
It's a little nit-picky (and I think I did this in some places too) but it's usually frowned upon to use bools as params to functions so if it is necessary to parameterize this function I think it might be cleaner to use an `enum class` something like

```cpp
enum class CheckOpts { Space, FrameRateDur, All }

void MainWindow::checkStartAcqRequirements(CheckOpts opts) {
    switch (opts) {
        case CheckOpts::Space:
               //do space check
               break;
        case CheckOpts::FrameRateDur:
              //do frame rate check
              break;
        case CheckOpts::All:
            //do all checks
            break;
    };
}

...
checkStartAcqRequirements(CheckOpts::Space);
...
checkStartAcqRequirements(CheckOpts::All);
```

Reply:
Or you could use bit flags defined as an enum, something like

```cpp
enum CheckOpts {
    SPACE = 0x01,
    FRAMERATE_DUR = 0x02,
}

void MainWindow::checkStart(uint opts) {
    ...
}

...
checkStart(SPACE | FRAMERATE_DUR);
...
checkStart(SPACE);
...
```

Reply:
c++20 also has designated initializers, which is something like

```cpp
struct CheckOpts {
    bool space = false;
    bool framerate_dur = false;
}

void check(CheckOpts opts) {
    ...
}

...
check({.space = true, .framerate_dur = true});
...
check({.space = true, .framerate_dur = false});
...
```